crates/compiler/builtins/roc/Hash.roc

```diff
@@ -66,3 +71,13 @@ Hasher has
     ## Completes the hasher, extracting a hash value from its
     ## accumulated hash state.
     complete : a -> U64 | a has Hasher
+
+## Adds a string into a [Hasher] by hashing its UTF-8 bytes.
+hashStrBytes = \hasher, s ->
+    Str.walkUtf8WithIndex s hasher \accumHasher, byte, _ ->
+        addU8 accumHasher byte
+
+## Adds a list of [Hash]able elements to a [Hasher] by hashing each element.
+hashList = \hasher, lst ->
```

Comment:
general question, long term can we add specialization of this? for list of integer types we can do things faster. Also, if we already have a list of bytes, we really want to hash it with the list of bytes function.

Reply:
Just to clarify, technically for any auto-derived hash function where the type has no padding when put in a list and does not contain any opaque type or pointer type, we should be able to just view the entire list as a list of bytes and get huge performance gains.

Reply:
Possibly. It does break a little bit a model of how derived ability implementations work if we go from generating derived impls for `List a` to `List U8` (or, anywhere where a is a concrete type), because of how lambda sets behave. I think for `Hash` it should be fine, but I'll need to think about it a bit more.

In the mean time, I don't think this is a problem is the `Hasher` implementer batches their work, as mentioned in the other review comment here.